source/core/TDCodecVia.cpp

```diff
@@ -1770,6 +1770,7 @@ TDViaFormatter::TDViaFormatter(StringRef string, Boolean quoteOnTopString, Int32
         _options._bEscapeStrings = false;
         _options._fmt = formatVIA;
     } else if (format->ComparePrefixCStr(formatJSON._name)) {
+        _options._precision = 17;
```

Comment:
I think you should use 16 if it works correctly.
Notice a lot of the constants that had to change below introduce inaccuracy because 17 is slightly more accurate than representable. e.g.  2.8 becomes  2.7999999999999998.